compiler/src/iree/compiler/Dialect/Modules/VMVX/Transforms/DecomposeLinalgGeneric.cpp

```diff
@@ -0,0 +1,220 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Dialect/Modules/VMVX/Transforms/PassDetail.h"
+#include "iree/compiler/Dialect/Modules/VMVX/Transforms/Passes.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+#include "mlir/IR/Matchers.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Pass/PassRegistry.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+namespace mlir {
+namespace iree_compiler {
+namespace IREE {
+namespace VMVX {
+
+namespace {
+
+struct DecomposeParallelGeneric
+    : public OpInterfaceRewritePattern<linalg::LinalgOp> {
+  using OpInterfaceRewritePattern::OpInterfaceRewritePattern;
+  struct BodyValueInfo {
+    // The generic operand that this body value maps to.
+    Value outerValue;
+    // The indexing map associated with this body value.
+    AffineMap indexingMap;
+  };
+  struct OpInfo {
+    linalg::LinalgOp op;
+
+    // The iteration domain, either as statically known values or dynamically
+    // known symbols. This will equal the number of loops and the number of
+    // dimensions in each indexing map.
+    SmallVector<OpFoldResult> domain;
```

Comment:
It looks like `domain` is only used by the size, and if IIUC size of `domainTypeDims` and `domain` is the same?

Reply:
Yes. Going to leave for now since we will be deleting all of this on the next integrate.